src/core/mac/data_poll_handler.cpp

```diff
@@ -31,17 +31,18 @@
  *   This file includes the implementation for handling of data polls and indirect frame transmission.
  */
 
-#if OPENTHREAD_FTD
-
 #include "data_poll_handler.hpp"
 
 #include "common/code_utils.hpp"
 #include "common/instance.hpp"
 #include "common/locator-getters.hpp"
 #include "common/logging.hpp"
+#include "thread/s2s.hpp"
 
 namespace ot {
 
+#if OPENTHREAD_FTD || OPENTHREAD_MTD_S2S
```

Comment:
Note that `OPENTHREAD_MTD_S2S` is defined in `thread/s2s.hpp` as a alias. (Otherwise we need to write `OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_SED_TO_SED_ENABLE` everywhere). Unlike other core config options, `OPENTHREAD_MTD_S2S` must be used after including `thread/s2s.hpp`. Thoughts?

Reply:
Got it. Thanks!